### PR TITLE
New way of handling google fonts inside the editor and frontend. Removed custom selectors in global typography

### PR DESCRIPTION
**Pull request by blankart**

### CODE REVIEW

File: src/fonts.php [hunk 1]
```diff
@@ -9,36 +9,30 @@
...
-            add_action( 'wp_head', array( $this, 'enqueue_frontend_block_fonts' ), 100 );
-		}
+		public static $google_fonts = [];
 
-		public function enqueue_frontend_block_fonts() {
-			if ( ! apply_filters( 'stackable_enqueue_fonts', true ) ) {
-				return;
-			}
+		function __construct() {
+				// add_action( 'wp_head', array( $this, 'enqueue_frontend_block_fonts' ), 100 );
```
Comment (bfintal) [inline src/fonts.php:17]:
```suggestion
```

File: src/fonts.php [hunk 2]
```diff
@@ -49,35 +42,17 @@ public function is_stackable_block( $block_name ) {
...
+				return;
+			}
 
-				// Look for fonts in inner blocks.
-				if ( ! empty( $block['innerBlocks'] ) ) {
-					$google_fonts = array_unique( array_merge( $google_fonts, $this->gather_google_fonts( $block['innerBlocks'] ) ) );
+			if ( ! in_array( $font_name, self::$google_fonts ) ) {
+				// Allow themes to disable enqueuing fonts, helpful for custom fonts.
+				if ( apply_filters( 'stackable_enqueue_font', true, $font_name ) ) {
+					self::$google_fonts[] = $font_name;
```
Comment (bfintal) [inline src/fonts.php:53]: let's just add this if the element doesn't exist yet in the array

File: src/global-settings.php [hunk 3]
```diff
@@ -54,7 +54,14 @@ function __construct() {
 			 * Typography hooks
 			 */
 
-			add_action( 'wp_enqueue_scripts', array( $this, 'typography_add_global_styles' ) );
+			/**
+			 * Uses `after_setup_theme` instead of `wp_enqueue_scripts` here for handling
+			 * page templates. Make sure to run `typography_add_global_styles` first before
+			 * the `typograhy_detect_native_blocks`  method.
```
Comment (bfintal) [inline src/global-settings.php:60]: I think this comment needs to be updated now.

```suggestion
			 * Use `after_setup_theme` to check early if there are global
			 * typography used the `typograhy_detect_native_blocks`  method.
```

File: src/higher-order/with-google-font/index.js [hunk 4]
```diff
@@ -6,24 +6,29 @@ import { loadGoogleFontInAttributes } from '~stackable/util'
...
-		static defaultProps = {
-			attributes: {},
-		}
+	WrappedComponent => {
+		const NewComponent = props => {
+			const isEditingTemplate = useSelect( select =>
+				select( 'core/edit-post' ).isEditingTemplate?.()
+			)
+
+			useEffect( () => {
```
Comment (bfintal) [inline src/higher-order/with-google-font/index.js:20]:
```suggestion
			// Reload the Google Fonts in the editor when switching to and from page template editing.
			useEffect( () => {
```

File: src/util/font.js [hunk 5]
```diff
@@ -31,21 +29,41 @@ export const isWebFont = fontName => ! fontName.match( /^(sans[-+]serif|serif|mo
...
  *
  * @param {string} fontName The name of the font
+ * @param customDoc
  */
 export const loadGoogleFont = fontName => {
-	if ( loadedFonts.includes( fontName ) ) {
-		return
-	}
+	setTimeout( () => {
+		const headElement = getDocumentHead()
```
Comment (bfintal) [inline src/util/font.js:36]:
```suggestion
```

File: src/util/font.js [hunk 6]
```diff
@@ -31,21 +29,41 @@ export const isWebFont = fontName => ! fontName.match( /^(sans[-+]serif|serif|mo
...
-		link.setAttribute( 'rel', 'stylesheet' )
-		link.setAttribute( 'type', 'text/css' )
-		document.querySelector( 'head' ).appendChild( link )
-	}
+				const link = createLinkTagWithGoogleFont( fontName )
+				head.appendChild( link )
+			}
+		}
+
+		_loadGoogleFont( headElement )
```
Comment (bfintal) [inline src/util/font.js:48]:
```suggestion
		const headElement = getDocumentHead()
		_loadGoogleFont( headElement )
```

File: src/util/font.js [hunk 7]
```diff
@@ -31,21 +29,41 @@ export const isWebFont = fontName => ! fontName.match( /^(sans[-+]serif|serif|mo
...
+		if ( headElement !== document.querySelector( 'head' ) ) {
+			_loadGoogleFont( document.querySelector( 'head' ) )
+		}
+	}, 50 )
+}
+
+export const createLinkTagWithGoogleFont = ( fontName = '' ) => {
+	const link = document.createElement( 'link' )
+	link.classList.add( `ugb-google-fonts` )
```
Comment (bfintal) [inline src/util/font.js:57]:
```suggestion
	link.classList.add( 'ugb-google-fonts' )
```

**Review by bfintal — CHANGES_REQUESTED** [review]
minor comments

File: src/fonts.php (hunk 2 above)
Comment (blankart) [reply src/fonts.php:53]: already handled

File: src/util/font.js [hunk 8]
```diff
@@ -31,21 +29,41 @@ export const isWebFont = fontName => ! fontName.match( /^(sans[-+]serif|serif|mo
  * Load the stylesheet of a Google Font.
  *
  * @param {string} fontName The name of the font
+ * @param customDoc
```
Comment (blankart) [inline src/util/font.js:32]:
```suggestion
 * @param {string} fontName The name of the font
```

Comment (blankart) [inline src/util/font.js:32]:
```suggestion
 * @param {string} fontName The name of the font
```

File: src/util/font.js [hunk 9]
```diff
@@ -31,21 +29,41 @@ export const isWebFont = fontName => ! fontName.match( /^(sans[-+]serif|serif|mo
...
  */
```
Comment (blankart) [inline src/util/font.js:33]:
```suggestion
 * Load the stylesheet of a Google Font.
 *
 * @param {string} fontName The name of the font
 */
```

**Pull request merged**